packages/flutter_tools/lib/src/project.dart

```diff
@@ -67,6 +67,57 @@ class FlutterProject {
   /// The generated IosModule sub project of this module project.
   IosModuleProject get iosModule => new IosModuleProject(directory.childDirectory('.ios'));
 
+  Future<File> get androidLocalPropertiesFile {
+    return _androidLocalPropertiesFile ??= manifest.then((FlutterManifest manifest) {
+      return directory.childDirectory(manifest.isModule ? '.android' : 'android')
+          .childFile('local.properties');
+    });
+  }
+  Future<File> _androidLocalPropertiesFile;
+
+  Future<File> get generatedXcodePropertiesFile {
+    return _generatedXcodeProperties ??= manifest.then((FlutterManifest manifest) {
+      return directory.childDirectory(manifest.isModule ? '.ios' : 'ios')
+          .childDirectory('Flutter')
+          .childFile('Generated.xcconfig');
+    });
+  }
+  Future<File> _generatedXcodeProperties;
+
+  File get flutterPluginsDotFile {
+    return _flutterPluginsDotFile ??= directory.childFile('.flutter-plugins');
+  }
+  File _flutterPluginsDotFile;
+
+  Future<Directory> get androidPluginRegistrantDirectory async {
```

Comment:
This `Directory` is not where we should place the registrant. It is the Android sub project in which we should place the registrant. This discrepancy might be confusing.

Maybe `androidPluginHost` or similar?

Reply:
Better!
Done